app/src/main/java/com/github/sdpcoachme/schedule/CreateEventActivity.kt

```diff
@@ -151,487 +169,646 @@ class CreateEventActivity : ComponentActivity() {
 
     private lateinit var store: CachingStore
     private lateinit var email: String
+    private lateinit var addressAutocompleteHandler: AddressAutocompleteHandler
+    private lateinit var selectSportsHandler: (Intent) -> CompletableFuture<List<Sports>>
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         store = (application as CoachMeApplication).store
         email = store.getCurrentEmail().get(1000, TimeUnit.MILLISECONDS)
```

Comment:
The email should be handled asynchronously and should not block the UI thread. I see that you did not modify this code, but it should be better handled just like in other activities.

Reply:
Yes actually, we don't even need the email in that activity anymore. I will take it out.